implementation/parallel-to-sns-scatter-gather/README.md

```diff
@@ -0,0 +1,130 @@
+```
+
+At this point you can now deploy the stacks.
+
+``` bash
+cdk deploy --all
+```
+
+To add additional dependencies, for example other CDK libraries, just add
+them to your `setup.py` file and rerun the `pip install -r requirements.txt`
+command.
+
+### Useful commands
+
+* `cdk ls`          list all stacks in the app
+* `cdk synth`       emits the synthesized CloudFormation template
+* `cdk deploy`      deploy this stack to your default AWS account/region
+* `cdk diff`        compare deployed stack with current state
+* `cdk docs`        open CDK documentation
+
+## Testing it out
+
+* First, lets invoke the step function workflow. Get the Step Function ARN from the Output of the cdk deploy step.(Get ARN from stack outputs: ```ScatterGatherWithParallelStack.StatemachineArn```)
+
+``` bash
+aws stepfunctions start-execution --state-machine-arn arn:aws:states:[aws_region]:[account-id]:stateMachine:ParallelStateForScatterGather  --input file://scatter_gather/input.json
+```
+
+You should see an output like that
+
+``` Json
+{
+    "executionArn": "arn:aws:states:us-east-1:446084752553:execution:ParallelStateForScatterGather:eef07f45-bbe8-4b36-9fdd-d96a4634b650",
+    "startDate": "2023-04-11T11:16:53.471000+02:00"
+}
+```
+
+* Next, lets test the refactored version that uses SNS and SQS to implement the Scatter-Gather patten. We will invoke the requester Lambda that triggers the execution. Get the function name from the cdk deploy output (Get function name from stack outputs:```ScatterGatherWithSNSStack.RequesterFunctionName```).
+
+``` bash
+aws lambda invoke-async --function-name ScatterGatherWithSNSStack-refactorlambdarequester8-3razgZDKZesx --invoke-args ./scatter_gather/input.json
+```
+
+You should see:
+
+``` Json
+{
+    "Status": 202
+}
+```
+
+*Note*: We used `invoke-async`  above because Lambda Destinations only supports Asynchronous invocation.
+
+## Verify
+
+Login to your AWS console and navigate to Step Function. Look into the ```ParallelStateForScatterGather``` state machine and verify the execution. Inspect the output of the aggregator step should look similar to the following.
+
+``` Json
+{
+  "statusCode": 200,
+  "body": "[{\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 616}, {\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 770}, {\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 330}, {\"uuid\": \"406ee76b-0f46-4191-9864-572dd153ccd4\", \"quote\": 847}]"
+}
+```
+
+Next, lets verify the refactored variant using SNS.
+First let's get the aggregator log group name.
+
+``` bash
+aws logs describe-log-groups --query 'logGroups[?starts_with(logGroupName, `/aws/lambda/ScatterGatherWithSNSStack-refactorlambdaaggregator-nfXGleA7rZsh`)].logGroupName' --output text
+```
+
+In the logstream you should see a line with the received quotes:
+
+``` bash
+aws logs tail /aws/lambda/ScatterGatherWithSNSStack-refactorlambdaaggregator-nfXGleA7rZsh  --filter-pattern "quotes" 
```

Comment:
Same as above, remind reader to replace lambda name to avoid `Function not found` error.